### PR TITLE
Change sandbox JWT format

### PR DESCRIPTION
**Pull request by rautenrieth-da**
This PR:
- Changes the payload format for the sandbox JWT tokens, wrapping any custom claims in a namespace field. This change was done to make the payload format OpenID Connect compatible.
- Changes the HTTP JSON API to expect the same tokens as the sandbox.

Both previous payload formats are deprecated, but still work.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-auth/src/test/suite/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTCodecSpec.scala [hunk 1]
```diff
@@ -48,6 +58,71 @@ class AuthServiceJWTCodecSpec
         minSuccessful(100))(
         value => serializeAndParse(value) shouldBe Success(value)
       )
+
+      "support legacy sandbox format" in {
```
Comment (gerolf-da) [inline ledger/ledger-api-auth/src/test/suite/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTCodecSpec.scala]: The new format with the OIDC namespace is handled by the `work for arbitrary values` test case?

**Comment by gerolf-da** [comment]
It could be interesting to log a deprecation warning when any of the legacy formats is being used. This would transport the signal that these formats aren't long for this world.

File: ledger/ledger-api-auth/src/test/suite/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTCodecSpec.scala (hunk 1 above)
Comment (rautenrieth-da) [reply ledger/ledger-api-auth/src/test/suite/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTCodecSpec.scala]: Correct, but I have now also added an explicit test for the new format.

**Comment by rautenrieth-da** [comment]
@leo I have fixed the JSON API documentation. FYI, the old payload format still works, but you will get a deprecation log warning:

```
WARN com.digitalasset.ledger.api.auth.AuthServiceJWTCodec$ - Token {"ledgerId":"someLedgerId","applicationId":"someApplicationId","party":"Alice"} is using a deprecated JWT payload format
```

**Pull request merged**